### PR TITLE
linux-pipewire: remove call to pw_deinit

### PR DESCRIPTION
**Pull request by kkartaltepe**
### Description
Per upstream this function is not supposed to be called by users of
pipewire. This also removes the nearly empty pipewire-common files.

ref: https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/2238#note_1311774

### Motivation and Context

May fix crashes on close for users in case anything else in their process calls pw_deinit. e.g. #5877

### How Has This Been Tested?

It compiled.

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: plugins/linux-pipewire/linux-pipewire.c [hunk 1]
```diff
@@ -53,5 +53,5 @@ bool obs_module_load(void)
 
 void obs_module_unload(void)
 {
-	obs_pipewire_unload();
+	// This space intentionally left blank.
```
Comment (GeorgesStavracas) [inline plugins/linux-pipewire/linux-pipewire.c:56]: Is this comment really necessary? I think we can go without :slightly_smiling_face:

**Review by GeorgesStavracas — CHANGES_REQUESTED** [review]
Please change the commit message capitalization to `linux-pipewire: Remove call to pw_deinit`

**Review by RytoEX — CHANGES_REQUESTED** [review]
Commit message nitpicks:
1. Write "PipeWire" when possible. The commit prefix need not comply with this.
2. Use sentence case for the first word after the commit prefix (capitalize the first letter).

**Comment by columbarius** [comment]
The PipeWire documenation was updated again:
> Since 0.3.49 this function must be paired with an equal amount of pw_init() calls to deinitialize the PipeWire library.

https://gitlab.freedesktop.org/pipewire/pipewire/-/blob/master/src/pipewire/pipewire.c#L681

Should we still remove `pw_deinit` or just hide it when compiled with PipeWire versions < 0.3.49?

**Comment by kkartaltepe** [comment]
> Should we still remove pw_deinit or just hide it when compiled with PipeWire versions < 0.3.49?

Does pipewire provide an appropriate macro to test this? We can add it back in if we want to have a clean(er) memsan/valgrind

**Comment by columbarius** [comment]
PipeWire has the macro `PW_CHECK_VERSION(major,minor,micro)`
https://gitlab.freedesktop.org/pipewire/pipewire/-/blob/master/src/pipewire/version.h.in#L59

**Comment by GeorgesStavracas** [comment]
Interesting, thanks @columbarius 

I guess we can go with

```c
void obs_pipewire_unload(void)
{
#if PW_CHECK_VERSION(0,3,50)
	pw_deinit();
#endif
}
```

**Comment by columbarius** [comment]
LGTM. I had plans to add additional stuff to pipewire-common when unifying common code between the plugins, but the changes just made things more complicated and they are probably lost for good.

**Comment by kkartaltepe** [comment]
Thanks, i set the version check to 0.3.49 since that should be the first safe version

**Review by GeorgesStavracas — CHANGES_REQUESTED** [review]
There's one extra `pipewire` in the commit message that should be changed to `PipeWire`.

LGTM other than that

**Review by GeorgesStavracas — APPROVED** [review]

**Pull request merged**